conans/client/subsystems.py

```diff
@@ -17,7 +17,7 @@ def run_in_windows_bash(conanfile, command, cwd=None, env=None):
     from conan.tools.env import Environment
     from conan.tools.env.environment import environment_wrap_command
     """ Will run a unix command inside a bash terminal It requires to have MSYS2, CYGWIN, or WSL"""
-    if env:
+    if env and env != "conanbuild":
```

Comment:
I still don't fully understand this. If the user explicitly pass ``self.run(env=None)``, maybe ``conanbuild.xxx`` shouldnt be applied, even inside the subsystem?